pages/api/checkout/session.ts

```diff
@@ -28,9 +28,31 @@ export default async function sessionHandler(
                 success_url: `${req.headers.origin}/result?session_id={CHECKOUT_SESSION_ID}`,
                 cancel_url: `${req.headers.origin}/checkout`,
                 allow_promotion_codes: true,
-            });
+            };
 
-            res.status(200).json({ sessionId: session.id });
+            // if a coupon ID is not passed in then allow promotion codes to be entered
+            if (!stripeRequest.couponId) {
+                const session = await stripe.checkout.sessions.create(
+                    stripeSession,
+                );
+
+                res.status(200).json({ sessionId: session.id });
+            }
+            // otherwise apply the discount from the coupon ID passed in
+            else {
+                delete stripeSession["allow_promotion_codes"];
+                stripeSession["discounts"] = [
+                    {
+                        coupon: stripeRequest.couponId,
+                    },
+                ];
+
+                const session = await stripe.checkout.sessions.create(
```

Comment:
Related to comment above! Since it's the same code, we can move it after the if else.

Reply:
```suggestion
```